crates/store/src/db/migrations/2025062000000_setup/up.sql

```diff
@@ -63,12 +64,15 @@ CREATE TABLE notes (
     PRIMARY KEY (committed_at, batch_index, note_index),
     CONSTRAINT notes_type_in_enum CHECK (note_type BETWEEN 1 AND 3),
     CONSTRAINT notes_execution_mode_in_enum CHECK (execution_mode BETWEEN 0 AND 1),
+    CONSTRAINT notes_transaction_id_is_digest CHECK (length(transaction_id) = 32),
     CONSTRAINT notes_consumed_at_is_u32 CHECK (consumed_at BETWEEN 0 AND 0xFFFFFFFF),
     CONSTRAINT notes_batch_index_is_u32 CHECK (batch_index BETWEEN 0 AND 0xFFFFFFFF),
     CONSTRAINT notes_note_index_is_u32 CHECK (note_index BETWEEN 0 AND 0xFFFFFFFF)
 );
 
 CREATE INDEX idx_notes_note_id ON notes(note_id);
+CREATE INDEX idx_notes_transaction_id ON notes(transaction_id);
+CREATE INDEX idx_notes_tx_join ON notes(sender, committed_at, transaction_id);
```

Comment:
This shouldn't be necessary with the suggestion https://github.com/0xMiden/miden-node/pull/1447/changes#r2641977678